### ISSUE TITLE
Basic authentication

### ISSUE THREAD

**Issue by PatricioDanos** [Azure/application-gateway-kubernetes-ingress#427]
While doing our dev for our application, it's common for us to do some basic auth to protect access to it.
Nginx ingress controller has Basic Auth as part of its solution by adding an htpasswd file as a secret, and a few annotations pointing to it. Here's the README.md for that implementation:

https://github.com/kubernetes/ingress-nginx/tree/master/docs/examples/auth/basic

Is it possible to implement something like this? This way we can secure access on a basic level until we implement a more robust authentication in our application code.

**Comment by Baklap4** [comment]
@delqn Any idea how we can easily achieve this? Basic authentication shouldn't be a showstopper to use the application gateway imho..

**Comment by PatricioDanos** [comment]
I've ended up removing this ingress controller and using nginx with modsecurity. It would be nice to have this feature in this ingress controller though.

**Comment by LucaPrete** [comment]
Let me echo this.
This is preventing us from using the application gateway controller where we do out at the edge.

What I'm personally looking at is the equivalent of these three annotations:

```
nginx.ingress.kubernetes.io/auth-tls-verify-client: "on"
nginx.ingress.kubernetes.io/auth-tls-verify-depth: "2"
nginx.ingress.kubernetes.io/auth-tls-secret: "XXX"
```

Any plan to implement this?

**Comment by akshaysngupta** [comment]
@LucaPrete We are close on supporting mutual auth on Application Gateway. Once that is ready, we will work on exposing it through AGIC.

**Comment by MarcelT-NL** [comment]
> @LucaPrete We are close on supporting mutual auth on Application Gateway. Once that is ready, we will work on exposing it through AGIC.
@akshaysngupta Are you saying client certificates will be supported on Appplication Gateway? That would be really great...?

**Comment by xs2bharat** [comment]
@akshaysngupta Is there any update on exposing mutual auth through AGIC as i understand that application gateway do support mutual auth via SSL profiles?

**Comment by jithin03** [comment]
Does AGIC has this basic auth feature like what Nginx provides

**Comment by sintetico82** [comment]
Hello, still working in progress?

**Comment by josedev-union** [comment]
> @LucaPrete We are close on supporting mutual auth on Application Gateway. Once that is ready, we will work on exposing it through AGIC.

@akshaysngupta Could you update the progress here?

**Comment by dash042** [comment]
it has been a couple of years - what is going on? everything OK?

**Comment by isujtauke** [comment]
> it has been a couple of years - what is going on? everything OK?

We transitioned all new sites to ingress-nginx due to things like this and app gateways having awful certificate limits.

**Comment by dash042** [comment]
> > it has been a couple of years - what is going on? everything OK?
> 
> We transitioned all new sites to ingress-nginx due to things like this and app gateways having awful certificate limits.

makes sense - I use Traefik for this and several other very good reasons but the customer thought the AppGW was a grand idea... :/
